distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestCatalogValidation.java

```diff
@@ -1065,21 +920,85 @@ private ValidatableResponse executeOpenSearch(String format, String... query) {
-            throws IOException {
-        if (checkResponse) {
-            return ingestXmlFromResource(resourceName);
+        List<String> ids = new ArrayList<>();
+        String id;
+        expect("Ingest failed to meet expectations").within(10, TimeUnit.SECONDS)
+                .checkEvery(1, TimeUnit.SECONDS)
+                .until(() -> {
+                    try {
+                        ids.add(ingest(writer.toString(), "text/xml", true));
+                        return !shouldFail;
+                    } catch (Error e) {
+                        return shouldFail;
+                    }
+
+                });
+        if (shouldFail) {
+            ids.stream()
+                    .forEach(mcardId -> safeDelete(mcardId));
+            return null;
         } else {
-            StringWriter writer = new StringWriter();
-            IOUtils.copy(IOUtils.toInputStream(getFileContent(resourceName)), writer);
-            return ingest(writer.toString(), "text/xml", checkResponse);
+            id = ids.get(0);
         }
+        expect("Failed to ingest xml from resource").within(10, TimeUnit.SECONDS)
+                .checkEvery(1, TimeUnit.SECONDS)
+                .until(() -> {
+                    try {
+                        String query = new CswQueryBuilder().addAttributeFilter(PROPERTY_IS_LIKE,
+                                "AnyText",
+                                "*")
+                                .getQuery();
+                        ValidatableResponse response = given().header(HttpHeaders.CONTENT_TYPE,
+                                MediaType.APPLICATION_XML)
+                                .body(query)
+                                .post(CSW_PATH.getUrl())
+                                .then();
+
+                        // Assert Metacard1 is in results AND Metacard2 because showInvalidMetacards is true
+                        response.body(hasXPath(format(
+                                "/GetRecordsResponse/SearchResults/Record[identifier=\"%s\"]",
+                                id)));
+                        return true;
+                    } catch (Error e) {
+                        LOGGER.warn("Ingest attempt failed. Will retry until timeout is reached.", e);
+                        return false;
+                    }
+
+                });
+        return id;
+    }
+
+    protected String ingestXmlFromResource(String resourceName) throws IOException {
+        return ingestXmlFromResource(resourceName, false);
+    }
+
+    private void safeDelete(String id) {
+        deleteMetacard(id);
+        expect("Failed to delete metacard within allotted time").within(10, TimeUnit.SECONDS)
+                .checkEvery(1, TimeUnit.SECONDS)
+                .until(() -> {
+                    try {
+                        String query = new CswQueryBuilder().addAttributeFilter(PROPERTY_IS_LIKE,
+                                "id",
```

Comment:
This logic has been replaced with a method that does a wildcard query and then looks for the id in the results. This should be safe.